Port types relating to SVG and ContentRule's to the new IPC serialization format

gavin-apple opened the pull request:
#### 1bf71c6c8b3007c849bcfbf1ebcb03581d9271a3
<pre>
Port types relating to SVG and ContentRule&apos;s to the new IPC serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=251864">https://bugs.webkit.org/show_bug.cgi?id=251864</a>
rdar://105133824

Reviewed by Alex Christensen.

Continues port of IPC types to the new serialization format. This change
includes:
    - SVGFilterExpressionNode
    - SVGPreserveAspectRatioValue
    - ContentRuleListResults::Result
    - ContentRuleListResults::Summary
    - ContentRuleListResults

* Source/WebCore/contentextensions/ContentRuleListResults.h:
(WebCore::ContentRuleListResults::Result::shouldNotifyApplication const):
(WebCore::ContentRuleListResults::shouldNotifyApplication const):
(WebCore::ContentRuleListResults::encode const): Deleted.
(WebCore::ContentRuleListResults::decode): Deleted.
(WebCore::ContentRuleListResults::Result::encode const): Deleted.
(WebCore::ContentRuleListResults::Result::decode): Deleted.
(WebCore::ContentRuleListResults::Summary::encode const): Deleted.
(WebCore::ContentRuleListResults::Summary::decode): Deleted.
* Source/WebCore/svg/SVGPreserveAspectRatioValue.h:
(WebCore::SVGPreserveAspectRatioValue::encode const): Deleted.
(WebCore::SVGPreserveAspectRatioValue::decode): Deleted.
* Source/WebCore/svg/graphics/filters/SVGFilterExpressionReference.h:
(WebCore::SVGFilterExpressionNode::encode const): Deleted.
(WebCore::SVGFilterExpressionNode::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: <a href="https://commits.webkit.org/260604@main">https://commits.webkit.org/260604@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e22023032e6e17ba61ae1aa0f1be27ab04609330

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108848 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17951 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41684 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/385 "Updated wpe dependencies (failure)") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/118133 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/112730 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19409 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9230 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101085 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114618 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/14554 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97767 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/42641 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/96534 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29410 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/84387 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10726 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30757 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11481 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/7689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16870 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50358 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7325 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13068 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/contentextensions/ContentRuleListResults.h:
```diff
@@ -52,20 +52,14 @@ struct ContentRuleListResults {
-
-        template<class Encoder> void encode(Encoder&) const;
-        template<class Decoder> static std::optional<Summary> decode(Decoder&);
+        Vector<ContentExtensions::ModifyHeadersAction> modifyHeadersActions { };
```
It would probably be worth adding a comment saying that these two are not serialized because _WKContentRuleListAction doesn't need them.

gavin-apple committed:
Port types relating to SVG and ContentRule's to the new IPC serialization format
https://bugs.webkit.org/show_bug.cgi?id=251864
rdar://105133824

Reviewed by Alex Christensen.

Continues port of IPC types to the new serialization format. This change
includes:
    - SVGFilterExpressionNode
    - SVGPreserveAspectRatioValue
    - ContentRuleListResults::Result
    - ContentRuleListResults::Summary
    - ContentRuleListResults

* Source/WebCore/contentextensions/ContentRuleListResults.h:
(WebCore::ContentRuleListResults::Result::shouldNotifyApplication const):
(WebCore::ContentRuleListResults::shouldNotifyApplication const):
(WebCore::ContentRuleListResults::encode const): Deleted.
(WebCore::ContentRuleListResults::decode): Deleted.
(WebCore::ContentRuleListResults::Result::encode const): Deleted.
(WebCore::ContentRuleListResults::Result::decode): Deleted.
(WebCore::ContentRuleListResults::Summary::encode const): Deleted.
(WebCore::ContentRuleListResults::Summary::decode): Deleted.
* Source/WebCore/svg/SVGPreserveAspectRatioValue.h:
(WebCore::SVGPreserveAspectRatioValue::encode const): Deleted.
(WebCore::SVGPreserveAspectRatioValue::decode): Deleted.
* Source/WebCore/svg/graphics/filters/SVGFilterExpressionReference.h:
(WebCore::SVGFilterExpressionNode::encode const): Deleted.
(WebCore::SVGFilterExpressionNode::decode): Deleted.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:

Canonical link: https://commits.webkit.org/260604@main
